src/calendar.rs

```diff
@@ -0,0 +1,190 @@
+            .delete(calendar_id)
+            .doit()
+            .await
+        {
+            error!(?why, "Failed to delete the calendar");
+            return Err(Error::GoogleError(why));
+        };
+        Ok(())
+    }
+
+    #[instrument(skip(self))]
+    pub async fn create_event(&self, event: Event, calendar_id: &str) -> Result<Event, Error> {
+        match self
+            .calendar_hub
+            .events()
+            .insert(event, calendar_id)
+            .doit()
+            .await
+        {
+            Ok(event) => Ok(event.1),
+            Err(why) => {
+                error!(?why, "Failed to create an event");
+                Err(Error::GoogleError(why))
+            }
+        }
+    }
+
+    #[instrument(skip(self))]
+    pub async fn delete_event(&self, id: &str, calendar_id: &str) -> Result<(), Error> {
+        if let Err(why) = self
+            .calendar_hub
+            .events()
+            .delete(calendar_id, id)
+            .doit()
+            .await
+        {
+            error!(?why, "Failed to delete an event");
+            return Err(Error::GoogleError(why));
+        };
+        Ok(())
+    }
+
+    #[instrument(skip(self))]
+    pub async fn list_calendars(&self) -> Result<Vec<CalendarListEntry>, Error> {
+        match self.calendar_hub.calendar_list().list().doit().await {
+            Ok(response) => Ok(response.1.items.unwrap_or_else(|| unreachable!())),
+            Err(why) => {
+                error!(?why, "Failed to list calendars");
+                Err(Error::GoogleError(why))
+            }
+        }
+    }
+
+    #[instrument(skip(self))]
+    pub async fn list_events(&self, calendar_id: &str) -> Result<Vec<Event>, Error> {
+        match self.calendar_hub.events().list(calendar_id).doit().await {
+            Ok(response) => Ok(response.1.items.unwrap_or_else(|| unreachable!())),
+            Err(why) => {
+                error!(?why, "Failed to list events");
+                Err(Error::GoogleError(why))
+            }
+        }
+    }
+
+    #[instrument(skip(self))]
+    pub async fn get_calendars_by_guild_id(
+        &self,
+        guild_id: &GuildId,
+    ) -> Result<Vec<CalendarListEntry>, Error> {
+        let mut response = self.list_calendars().await?;
+        response.retain(|calendar| calendar.summary == Some(guild_id.to_string()));
+        if response.len() > 1 {
+            warn!("More than one calendar associated with the server, using the newest one");
+        };
+        Ok(response)
+    }
+
+    #[instrument(skip(self))]
+    pub async fn get_event_id_by_label(
+        &self,
+        label: &str,
+        calendar_id: &str,
+    ) -> Result<Vec<String>, Error> {
+        let response = self.list_events(calendar_id).await?;
+        let value = response
+            .into_iter()
+            .filter_map(|event| {
+                if event.summary == Some(label.to_string()) {
+                    Some(event.id.unwrap_or_else(|| unreachable!()))
```

Comment:
Is this really unreachable? I would think not. 

Use an `.expect()` here if you're sure that the value would always be present.

Reply:
Yes, that's unreachable as every event has an id. I think it is possible only if there's any problems on Google side, but it will rather return an Error, than an empty id.

Reply:
My point being that it's still reachable if the service returns an invalid response. 

In either case, .expect is the idiom for "I am so sure that this is going to be here i'm willing to crash the program for it".

The unreachable macro is more of a compiler optimization.